CONTRIBUTING.md

```diff
@@ -1,50 +1,66 @@
 # Contributing to SlimeVR
 
-This document describes essential knowledge for contributors to SlimeVR.
+This document describes essential knowledge required to contribute to the SlimeVR Server.
 
-## How to get started
+### Prerequisites
 
-### Getting the code
-First, clone the codebase using `git`. If you don't have `git` installed, go install it.
+- [Git](https://git-scm.com/downloads)
+- [Java v17+](https://adoptium.net/temurin/releases/)
+- [Node.js v16+](https://nodejs.org) (We recommend the use of `nvm` instead of installing Node.js directly)
+- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or ``webkit2gtk`` for Linux
+- [Rust](https://rustup.rs)
+
+## Cloning the code
+First, clone the codebase using git in a terminal in the folder you want.
 
 ```bash
-# Clone repositories
 git clone --recursive https://github.com/SlimeVR/SlimeVR-Server.git
-
-# Enter the directory of the codebase
-cd SlimeVR-Server
 ```
 
-Now you can open the codebase in your favorite IDE or text editor.
+Now you can open the codebase in [IDEA](https://www.jetbrains.com/idea/download/) (Recommended; VSCode and Eclipse also work but have limited Kotlin support).
 
-### Installing Java
-The codebase is required to build with Java version 17 or higher.
 
-```bash
-# Check java version
-java --version
-```
+## Building the code
+
+### Java (server)
 
+The Java code is built with `gradle`, a CLI tool that manages java projects and their
+dependencies.
+- You can run the server by running `./gradlew run` in your IDE's terminal.
+- To compile the code, run `./gradlew shadowJar`. The result will
+be at `server/build/libs/slimevr.jar` (you can ignore `server.jar`).
 
-### Building the code
-The code is built with `gradle`, a cli tool that manages java projects and their
-dependencies. You can build the code with `./gradlew build` and run it with
-`./gradlew run`.
+(Note: your IDE may do all of the above for you.)
 
+### Tauri (gui)
 
-## Code Style
-Code is autoformatted with [spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle).
+- Run `npm install` in your IDE's terminal to download and install dependencies.
+- To launch the GUI in dev mode, run `npm run gui`.
```

Comment:
```suggestion
- To launch the GUI in dev mode, run `npm run tauri dev`.
```
Shouldn't this be like this instead? idk what `npm run gui` does.

Reply:
`npm run gui` runs `npm i solarxr-protocol`, so it's better to avoid confusion and extra steps.